module/pattern/launch/src/main/java/org/openbase/jul/pattern/launch/AbstractLauncher.java

```diff
@@ -627,18 +436,245 @@ private static void printSummary(final Class application, final Logger logger, f
+
+            synchronized (LAUNCHER_LOCK) {
+
+                setState(LauncherState.INITIALIZING);
+
+                try {
+                    init();
+                    activate();
+                } catch (CouldNotPerformException ex) {
+                    ExceptionPrinter.printHistory("Could not activate Launcher[" + getName() + "]!", ex, logger);
+                } catch (RuntimeException e) {
+                    setState(LauncherState.ERROR);
+                    throw e;
+                } catch (InterruptedException e) {
+                    setState(LauncherState.STOPPED);
+                    throw e;
+                }
+
+                launchable = instantiateLaunchable();
+                try {
+                    launchable.init();
+                    setState(LauncherState.LAUNCHING);
+
+                    launchable.activate();
+                    launchTime = System.currentTimeMillis();
+                    setState(LauncherState.RUNNING);
+                    try {
+                        verify();
+                        verified = true;
+                    } catch (VerificationFailedException ex) {
+                        verified = false;
+                        verificationFailedException = ex;
+                    }
+                } catch (InterruptedException ex) {
+                    setState(LauncherState.STOPPING);
+                    return null;
+                } catch (Exception ex) {
+                    setState(LauncherState.ERROR);
+                    launchable.shutdown();
+                    if (!ExceptionProcessor.isCausedBySystemShutdown(ex)) {
+                        ExceptionPrinter.printHistoryAndReturnThrowable(new CouldNotPerformException("Could not launch " + getName(), ex), logger);
+                    }
+                }
+                return null;
+            }
+        });
+        return launcherTask;
+    }
+
+    @Override
+    public void relaunch() throws CouldNotPerformException, InterruptedException {
+        synchronized (LAUNCHER_LOCK) {
+            stop();
+            try {
+                launch().get();
+            } catch (ExecutionException | CancellationException ex) {
+                throw new CouldNotPerformException(ex);
+            }
+        }
+    }
+
+    @Override
+    public void stop() {
+
+        interruptBoot();
+
+        synchronized (LAUNCHER_LOCK) {
+            setState(LauncherState.STOPPING);
+            if (launchable != null) {
+                launchable.shutdown();
+            }
+            setState(LauncherState.STOPPED);
+        }
+    }
+
+    /**
+     * Method cancels the boot process of this launcher.
+     */
+    private void interruptBoot() {
+        if (isBooting()) {
+            launcherTask.cancel(true);
+        }
+    }
+
+    /**
+     * @return true if the launcher is currently booting.
+     */
+    private boolean isBooting() {
+        return launcherTask != null && !launcherTask.isDone();
+    }
```

Comment:
I know this is not part of this PR but why aren't these methods called `interruptLaunch` and `isLaunching`?